kong/ids.go

```diff
@@ -103,6 +103,30 @@ func (cg *ConsumerGroup) FillID() error {
 	return nil
 }
 
+// FillID fills the ID of an entity. It is a no-op if the entity already has an ID.
+// ID is generated in a deterministic way using UUIDv5. The UUIDv5 namespace is different for each entity type.
+// The name used to generate the ID for Vault is Vault.Prefix.
+func (v *Vault) FillID() error {
+	if v == nil {
+		return fmt.Errorf("vault is nil")
+	}
+	if v.ID != nil && *v.ID != "" {
+		// ID already set, do nothing.
+		return nil
+	}
+	if v.Prefix == nil || *v.Prefix == "" {
```

Comment:
We could probably check this with `len(v.Prefix) == 0` as well with the same effect.

Reply:
Done.